### PR TITLE
Convert error strings to proper validation errors.

### PR DESCRIPTION
**Pull request by rjnagal**

### CODE REVIEW

**Comment by rjnagal** [comment]
@thockin
 fixes #6031

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -469,7 +469,7 @@ func ValidatePersistentVolumeStatusUpdate(newPv, oldPv *api.PersistentVolume) er
 	allErrs := errs.ValidationErrorList{}
 	allErrs = append(allErrs, ValidateObjectMetaUpdate(&oldPv.ObjectMeta, &newPv.ObjectMeta).Prefix("metadata")...)
 	if newPv.ResourceVersion == "" {
-		allErrs = append(allErrs, fmt.Errorf("ResourceVersion must be specified"))
+		allErrs = append(allErrs, errs.NewFieldRequired("ResourceVersion"))
```
Comment (thockin) [inline pkg/api/validation/validation.go]: field names should be JSON style - resourceVersion

File: pkg/api/validation/validation.go [hunk 2]
```diff
@@ -1161,7 +1161,7 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 
 func validateBasicResource(quantity resource.Quantity) errs.ValidationErrorList {
 	if quantity.Value() < 0 {
-		return errs.ValidationErrorList{fmt.Errorf("%v is not a valid resource quantity", quantity.Value())}
+		return errs.ValidationErrorList{errs.NewFieldInvalid("", quantity.Value(), "must be a valid resource quantity")}
```
Comment (thockin) [inline pkg/api/validation/validation.go]: you really want a field name here - can you pass that down into this function?

File: pkg/api/validation/validation.go [hunk 3]
```diff
@@ -1242,12 +1242,12 @@ func ValidateNamespace(namespace *api.Namespace) errs.ValidationErrorList {
 func validateFinalizerName(stringValue string) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
 	if !util.IsQualifiedName(stringValue) {
-		return append(allErrs, fmt.Errorf("finalizer name: %v, %v", stringValue, qualifiedNameErrorMsg))
+		return append(allErrs, errs.NewFieldInvalid("spec.Finalizers", stringValue, qualifiedNameErrorMsg))
```
Comment (thockin) [inline pkg/api/validation/validation.go]: JSON style

File: pkg/api/validation/validation.go (hunk 2 above)
Comment (rjnagal) [inline pkg/api/validation/validation.go]: The field name is already added at the callsite through Prefix().

Comment (rjnagal) [inline pkg/api/validation/validation.go]: Updated all other fields.

**Pull request merged**